source/server/async_method_context.h

```diff
@@ -0,0 +1,83 @@
+#ifndef NIDEVICE_GRPC_DEVICE_ASYNC_METHOD_CONTEXT_H
+#define NIDEVICE_GRPC_DEVICE_ASYNC_METHOD_CONTEXT_H
+
+#include <grpcpp/grpcpp.h>
+
+#include <memory>
+
+#include "async_server_write_queue.h"
+
+namespace nidevice_grpc {
+using AsyncOperationToken = void*;
+// Type erasing abstract base class to allow managing driver-specific operation
+// registration in an AsyncMethodContext.
+struct AsyncOperationRegistration {
+  virtual ~AsyncOperationRegistration() {}
+  virtual AsyncOperationToken token() { return reinterpret_cast<AsyncOperationToken>(0); }
+};
+
+struct AsyncMethodContext {
```

Comment:
Note: This baseclass/interface allows many of the operations to no longer be templatized. This means:
1. They can move to `async_operation.cpp`
2. They will no longer get unique instantiations for each method type.

Note that some operations (i.e., `WriteOperation`) still need `TResponse` or `TRequest` and will use `AsyncMethodContextT` below.